### DISCUSSION TITLE
Can starlette route to a unix domain socket?

### DISCUSSION THREAD

**Discussion by deleted** [encode/starlette#2150] [Q&A] [answered]
I have an existing GRPC server (built on [grpclib](https://github.com/vmagamedov/grpclib)), and I'd like to add a couple of traditional HTTP endpoints to the same service port, so I'm looking for a python webserver that can both serve it's own HTTP requests and also route to my GRPC server.  Since GRPC is HTTP/2, the server needs to be able to support HTTP/2 as well.

Grpclib's Server can bind to a [UNIX Domain Socket](https://en.wikipedia.org/wiki/Unix_domain_socket) instead of a host port. Does Starlette provide a way to route requests to a unix domain socket instead of an Endpoint instance?

**Comment by alex-oleshkevich** [comment, marked as answer]
Starlette is a web framework, not a server. Thus, it cannot be bound to Unix or TCP socket. You should use a dedicated server (like uvicorn for HTTP). For example, such a server can bind to the Unix socket and then route requests to Starlette.
